integration/test/appcatalog/appcatalogentry/app_catalog_entry_test.go

```diff
@@ -0,0 +1,159 @@
+			entryCR, err = config.K8sClients.G8sClient().ApplicationV1alpha1().AppCatalogEntries(metav1.NamespaceDefault).Get(ctx, key.AppCatalogEntryName(), metav1.GetOptions{})
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			return nil
+		}
+
+		n := func(err error, t time.Duration) {
+			config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("failed to get appcatalogentry CR with name %#q: retrying in %s", key.AppCatalogEntryName(), t), "stack", fmt.Sprintf("%v", err))
+		}
+
+		b := backoff.NewConstant(5*time.Minute, 15*time.Second)
+		err := backoff.RetryNotify(o, b, n)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+	}
+
+	{
+		expectedLabels := map[string]string{
+			pkglabel.AppKubernetesName: "prometheus-operator-app",
+			pkglabel.CatalogName:       key.StableCatalogName(),
+			pkglabel.CatalogType:       "stable",
+			pkglabel.Latest:            "false",
+			label.ManagedBy:            "app-operator-unique",
+		}
+
+		if !reflect.DeepEqual(entryCR.Labels, expectedLabels) {
+			t.Fatalf("want matching labels \n %s", cmp.Diff(entryCR.Labels, expectedLabels))
+		}
+
+		expectedEntrySpec := v1alpha1.AppCatalogEntrySpec{
+			AppName:    "prometheus-operator-app",
+			AppVersion: "0.38.1",
+			Catalog: v1alpha1.AppCatalogEntrySpecCatalog{
+				Name:      key.StableCatalogName(),
+				Namespace: "",
+			},
+			Chart: v1alpha1.AppCatalogEntrySpecChart{
+				Home: "https://github.com/giantswarm/prometheus-operator-app",
+				Icon: "https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/assets/prometheus_logo-cb55bb5c346.png",
+			},
+			DateCreated: nil,
+			DateUpdated: nil,
+			Version:     "0.3.4",
+		}
+
+		// Clear dates for comparison.
+		entryCR.Spec.DateCreated = nil
+		entryCR.Spec.DateUpdated = nil
+
+		if !reflect.DeepEqual(entryCR.Spec, expectedEntrySpec) {
+			t.Fatalf("want matching spec \n %s", cmp.Diff(entryCR.Spec, expectedEntrySpec))
+		}
+	}
+
+	{
+		err = config.K8sClients.G8sClient().ApplicationV1alpha1().AppCatalogs().Delete(ctx, key.StableCatalogName(), metav1.DeleteOptions{})
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		o := func() error {
+			lo := metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("%s=%s,%s=%s", label.ManagedBy, project.Name(), pkglabel.CatalogName, key.StableCatalogName()),
+			}
+			entryCRs, err := config.K8sClients.G8sClient().ApplicationV1alpha1().AppCatalogEntries(metav1.NamespaceDefault).List(ctx, lo)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+			if len(entryCRs.Items) > 0 {
+				return microerror.Maskf(testError, "expected 0 appcatalogentries got %d", len(entryCRs.Items))
+			}
+
+			return nil
+		}
+
+		n := func(err error, t time.Duration) {
+			config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("appcatalogentry CRs still exist: retrying in %s", t), "stack", fmt.Sprintf("%v", err))
+		}
+
+		b := backoff.NewConstant(5*time.Minute, 15*time.Second)
```

Comment:
Can we change it to `MaxRetry` if it would be reconciled by the event?

Reply:
Cool yes its triggered by the event so using max retry is probably faster.

Updated. PTAL